atomix/cluster/src/main/java/io/atomix/raft/cluster/impl/RaftClusterContext.java

```diff
@@ -50,30 +40,23 @@
 import java.util.concurrent.CopyOnWriteArraySet;
 import java.util.function.Consumer;
 import java.util.stream.Collectors;
-import org.slf4j.Logger;
 
 /** Manages the persistent state of the Raft cluster from the perspective of a single server. */
 public final class RaftClusterContext implements RaftCluster, AutoCloseable {
 
-  private final Logger log;
   private final RaftContext raft;
   private final DefaultRaftMember member;
   private final Map<MemberId, RaftMemberContext> membersMap = new ConcurrentHashMap<>();
   private final Set<RaftMember> members = new CopyOnWriteArraySet<>();
   private final List<RaftMemberContext> remoteMembers = new CopyOnWriteArrayList<>();
   private final Map<RaftMember.Type, List<RaftMemberContext>> memberTypes = new HashMap<>();
-  private final Set<RaftClusterEventListener> listeners = new CopyOnWriteArraySet<>();
   private volatile Configuration configuration;
-  private volatile Scheduled joinTimeout;
-  private volatile CompletableFuture<Void> joinFuture;
+  private volatile CompletableFuture<Void> bootstrapFuture;
```

Comment:
I can do. I actually wanted to set config to final, but it is overwritten in same cases. Can we talk about that? I'm wondering whether these are necessary. For example here configure is called https://github.com/zeebe-io/zeebe/blob/develop/atomix/cluster/src/main/java/io/atomix/raft/cluster/impl/DefaultRaftMember.java#L186 This is done on promote or denote a member to a certain state, like from PASSIVE to ACTIVE. 

Do we want to support the promote/demote stuff ? (\cc @npepinpe )

Reply:
I would propose that we remove these configuration changes as well.

Reply:
Ok let's do the following. I will use here atomic references instead of volatile and create a follow up issue to remove promote and demote, so we can make further progress and merge this PR already

Reply:
https://github.com/zeebe-io/zeebe/issues/6212